sklearn/cross_validation.py

```diff
@@ -1430,213 +1270,22 @@ def cross_val_score(estimator, X, y=None, scoring=None, cv=None, n_jobs=1,
-    if hasattr(estimator, 'kernel') and callable(estimator.kernel) \
-       and not isinstance(estimator.kernel, GPKernel):
-        # cannot compute the kernel values with custom function
-        raise ValueError("Cannot use a custom kernel function. "
-                         "Precompute the kernel matrix instead.")
-
-    if not hasattr(X, "shape"):
-        if getattr(estimator, "_pairwise", False):
-            raise ValueError("Precomputed kernels or affinity matrices have "
-                             "to be passed as arrays or sparse matrices.")
-        X_subset = [X[idx] for idx in indices]
-    else:
-        if getattr(estimator, "_pairwise", False):
-            # X is a precomputed square kernel matrix
-            if X.shape[0] != X.shape[1]:
-                raise ValueError("X should be a square kernel matrix")
-            if train_indices is None:
-                X_subset = X[np.ix_(indices, indices)]
-            else:
-                X_subset = X[np.ix_(indices, train_indices)]
-        else:
-            X_subset = safe_indexing(X, indices)
-
-    if y is not None:
-        y_subset = safe_indexing(y, indices)
-    else:
-        y_subset = None
-
-    return X_subset, y_subset
-
-
-def _score(estimator, X_test, y_test, scorer):
-    """Compute the score of an estimator on a given test set."""
-    if y_test is None:
-        score = scorer(estimator, X_test)
-    else:
-        score = scorer(estimator, X_test, y_test)
-    if not isinstance(score, numbers.Number):
-        raise ValueError("scoring must return a number, got %s (%s) instead."
-                         % (str(score), type(score)))
-    return score
-
-
-def _permutation_test_score(estimator, X, y, cv, scorer):
-    """Auxiliary function for permutation_test_score"""
-    avg_score = []
-    for train, test in cv:
-        estimator.fit(X[train], y[train])
-        avg_score.append(scorer(estimator, X[test], y[test]))
-    return np.mean(avg_score)
 
+# The order of the labels argument changes in the new module
+def permutation_test_score(estimator, X, y, cv=None,
+                           n_permutations=100, n_jobs=1, labels=None,
+                           random_state=0, verbose=0, scoring=None):
+    return model_selection.permutation_test_score(
+        estimator, X, y, labels=labels, cv=cv, n_permutations=n_permutations,
+        n_jobs=n_jobs, random_state=random_state, verbose=verbose,
+        scoring=scoring)
 
-def _shuffle(y, labels, random_state):
-    """Return a shuffled copy of y eventually shuffle among same labels."""
-    if labels is None:
-        ind = random_state.permutation(len(y))
-    else:
-        ind = np.arange(len(labels))
-        for label in np.unique(labels):
-            this_mask = (labels == label)
-            ind[this_mask] = random_state.permutation(ind[this_mask])
-    return y[ind]
+# Using docstring from model_selection
+permutation_test_score.__doc__ = model_selection.permutation_test_score.__doc__
```

Comment:
for the record, I don't think this works for sphinx, but we don't include the deprecated module on the website anyhow ;)